tests/modules/seq2seq_encoders/efficient_multi_head_attention_test.py

```diff
@@ -0,0 +1,33 @@
+# pylint: disable=invalid-name,no-self-use,too-many-public-methods
+import numpy
+import torch
+from torch.autograd import Variable
+
+from allennlp.common.testing import AllenNlpTestCase
+from allennlp.modules.seq2seq_encoders import EfficientMultiHeadAttention
+
+
+class EfficientMultiHeadAttentionTest(AllenNlpTestCase):
+    def test_multi_head_attention_runs_forward(self):
+        attention = EfficientMultiHeadAttention(num_heads=3,
+                                                input_dim=5,
+                                                attention_dim=7,
+                                                values_dim=9)
+        inputs = Variable(torch.randn(2, 12, 5))
+        attention(inputs)
+
+    def test_multi_head_attention_respects_masking(self):
+        attention = EfficientMultiHeadAttention(num_heads=3,
```

Comment:
Is there an easy way to test for the reshape issue you mentioned above?  Probably not...

Maybe if you set the weights by hand and gave some simple inputs...